mongo/mongo_test.go

```diff
@@ -342,6 +342,27 @@ func TestMongoHelpers(t *testing.T) {
 				true,
 				nil,
 			},
+			{
+				"semantic single document/bson.D",
+				bson.D{},
+				nil,
+				false,
+				errors.New("can only transform types that are semantically arrays of documents, got primitive.D"),
```

Comment:
The `%T` `fmt` verb interprets `bson.D` as `primitive.D` since `bson.D` is actually just a [type-alias](https://github.com/mongodb/mongo-go-driver/blob/master/bson/bson.go#L33) of `primitive.D`.